core/src/main/scala/org/apache/spark/status/LiveEntity.scala

```diff
@@ -510,6 +510,7 @@ private class LiveRDD(val info: RDDInfo) extends LiveEntity {
   var storageLevel: String = weakIntern(info.storageLevel.description)
   var memoryUsed = 0L
   var diskUsed = 0L
+  var storageInfo: StorageLevel = new StorageLevel()
```

Comment:
I don't know this part well, but is it redundant with `storageLevel` above?

Reply:
The above was just a string representation of storage level. from StorageInfo we can get individual parameters includes, replication

Reply:
I see, but should we not just replace the field above with this richer object? or should this not use `info.storageLevel` as the initial value? maybe not, just jumped out at me as a question

Reply:
Yes. we can initialize `storageInfo = info.storageLevel`. But I'm not sure we can get rid of `storageLevel`, as there is a public method which sets the value. updated the code.